src/components/PluginTabs.tsx

```diff
@@ -0,0 +1,208 @@
+import { ProbesPage } from 'page/ProbesPage';
+import { TenantSetup } from 'components/TenantSetup';
+import { InstanceContext } from './InstanceContext';
+import { getLocationSrv } from '@grafana/runtime';
+import { DashboardInfo } from 'datasource/types';
+import { listAppDashboards } from 'dashboards/loader';
+import { Button, HorizontalGroup, Modal } from '@grafana/ui';
+
+type Tab = {
+  label: string;
+  id: string;
+  icon?: string;
+  enabledByFeatureFlag?: string;
+  render: (query: any) => JSX.Element;
+};
+
+type RouterQuery = {
+  page: string;
+  id: string;
+};
+
+const pagesToRedirectIfNotInitialized = new Set(['checks', 'probes', 'config', 'redirect']);
+
+const pagesToRedirectIfInitialized = new Set(['setup']);
+
+const dashboardRedirects = new Set(['redirect']);
+
+const getRedirectDestination = (queryPage: string, isInitialized: boolean): string | undefined => {
+  if (!isInitialized && pagesToRedirectIfNotInitialized.has(queryPage)) {
+    return 'setup';
+  }
+  if (isInitialized && pagesToRedirectIfInitialized.has(queryPage)) {
+    return 'checks';
+  }
+  if (isInitialized && dashboardRedirects.has(queryPage)) {
+    return '';
+  }
+  return;
+};
+
+const tabs: Tab[] = [
+  {
+    label: 'Checks',
+    id: 'checks',
+    render: (query: RouterQuery) => <ChecksPage id={query.id} />,
+  },
+  {
+    label: 'Probes',
+    id: 'probes',
+    render: (query: RouterQuery) => <ProbesPage id={query.id} />,
+  },
+
+  {
+    label: 'Config',
+    id: 'config',
+    render: (query: RouterQuery) => <TenantSetup />,
+  },
+];
+
+function filterTabs(tabs: Tab[], apiInitialized: boolean): Tab[] {
+  if (!apiInitialized) {
+    return [];
+  }
+  return tabs;
+}
+
+function findActiveTab(tabs: Tab[], queryPage: string, apiInitialized: boolean): Tab {
+  return tabs.find(tab => tab.id === queryPage) ?? tabs[0];
+}
+
+function handleDashboardRedirect(dashboard: string, dashboards: DashboardInfo[]) {
+  const targetDashboard = dashboards.find(dashboardJson => dashboardJson.json.indexOf(dashboard) > -1) ?? dashboards[0];
+  getLocationSrv().update({
+    partial: false,
+    path: `d/${targetDashboard.uid}`,
+  });
+}
+
+function getNavModel(tabs: Tab[], path: string, activeTab: Tab, logoUrl: string) {
+  const children = tabs.map(tab => ({
+    text: tab.label,
+    id: tab.id,
+    active: tab.id === activeTab.id,
+    url: `${path}?page=${tab.id}`,
+  }));
+
+  const node = {
+    text: 'Synthetic Monitoring',
+    img: logoUrl,
+    subTitle: 'Grafana Cloud Synthetic Monitoring',
+    url: path,
+    children,
+  };
+  if (tabs.length) {
+    return {
+      main: node,
+      node,
+    };
+  }
+  return null;
+}
+
+export const PluginTabs: FC<AppRootProps<GlobalSettings>> = ({ query, onNavChanged, path, meta }) => {
+  const { instance } = useContext(InstanceContext);
+  const [hasDismissedDashboardUpdate, setHasDismissedDashboardUpdate] = useState(
+    Boolean(window.sessionStorage.getItem('hasDismissedDashboardUpdate'))
+  );
```

Comment:
**Idea:** these client-side-persistence related logic could possibly live in an other layer.

In that case it would be easier to oversee in the future actually what data are we persisting on the client-side, and it would also be easier to change where we would like to persist them and how.